include/itkTileMerging.h

```diff
@@ -0,0 +1,218 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef itkTileMerging_h
+#define itkTileMerging_h
+
+#include "itkTileMontage.h"
+#include "itkLinearInterpolateImageFunction.h"
+#include "itkSimpleFastMutexLock.h"
+
+namespace itk
+{
+/** \class TileMerging
+ * \brief Resamples an n-Dimensional mosaic of images into a single big image.
+ *
+ * CropToFill indicates whether the big image will be cropped so it
+ * entirely consists of input tiles (no default background filling).
+ * If CropToFill is false, the big image will have the extent to include
+ * all of the input tiles. The pixels not covered by any input tile
+ * will have the value specified by the Background member variable.
+ *
+ * \author Dženan Zukić, dzenan.zukic@kitware.com
+ *
+ * \ingroup Montage
+ */
+template <typename TImageType, typename TInterpolator = LinearInterpolateImageFunction<TImageType, float> >
+class ITK_TEMPLATE_EXPORT TileMerging: public TileMontage<TImageType, typename TInterpolator::CoordRepType>
+{
+public:
+  ITK_DISALLOW_COPY_AND_ASSIGN(TileMerging);
+
+  /** Standard class type aliases. */
+  using Self = TileMerging;
+  using Superclass = TileMontage<TImageType, typename TInterpolator::CoordRepType>;
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+  using ImageType = TImageType;
+  using ImagePointer = typename ImageType::Pointer;
+  using ImageConstPointer = typename ImageType::ConstPointer;
+
+  /** Method for creation through the object factory. */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(TileMerging, TileMontage);
+
+  /** Dimensionality of input images. */
+  itkStaticConstMacro(ImageDimension, unsigned int, ImageType::ImageDimension);
+
+  /** This is envisioned to be the primary way of setting inputs.
+   * All required inputs are taken from TileMontage. Alternatively,
+   * inherited members can be called individually, e.g.:
+   * SetMontageSize(), SetInputTile(), SetOriginAdjustment() etc. */
```

Comment:
Interesting. Do we need to inherit also?

Reply:
We reduce a lot of duplication by inheriting.